platform/admin/core/admin-core-insecuredefaults/src/main/java/org/codice/ddf/admin/insecure/defaults/service/UserPropertiesDeletionScheduler.java

```diff
@@ -0,0 +1,150 @@
+  private static boolean deletionScheduled;
+
+  public UserPropertiesDeletionScheduler() {
+    deletionScheduled = false;
+  }
+
+  public void scheduleDeletion(Path path) {
+    if (!deletionScheduled) {
+      String cron = getCron(path);
+      if (cron == null) {
+        return;
+      }
+      try {
+        CamelContext context = new DefaultCamelContext();
+        RouteBuilder builder =
+            new RouteBuilder() {
+              @Override
+              public void configure() throws Exception {
+                fromF("quartz2://scheduler/deletionTimer?cron=%s", cron)
+                    .bean(UserPropertiesDeletionScheduler.class, "deleteFiles");
+              }
+            };
+
+        context.addRoutes(builder);
+        context.start();
+        deletionScheduled = true;
+      } catch (Exception e) {
+        LOGGER.debug("Unable to create route", e);
+      }
+    }
+  }
+
+  @VisibleForTesting
+  String getCron(Path path) {
+    try {
+      if (!path.toFile().exists()) {
+        deleteFiles();
+        return null;
+      }
+
+      if (!TEMP_FILE_TIMESTAMP_PATH.toFile().exists()) {
+
+        File tempFile = new File(TEMP_FILE_TIMESTAMP_PATH.toString());
+        if (!tempFile.createNewFile()) {
+          return null;
+        }
+        Files.write(
+            TEMP_FILE_TIMESTAMP_PATH,
+            Long.toString(System.currentTimeMillis()).getBytes(StandardCharsets.UTF_8));
+
+        return cronCalculator(new Date(System.currentTimeMillis()));
+
+      } else {
+        Date firstInstall = installationDate();
+        if (firstInstall == null) {
+          return null;
+        }
+        Date timestamp = new Date(firstInstall.getTime() + TimeUnit.DAYS.toMillis(3));
+        Date currentDate = new Date();
+        if (currentDate.before(timestamp)) {
+          return cronCalculator(firstInstall);
+        }
+        deleteFiles();
+      }
+    } catch (IOException e) {
+      LOGGER.debug("Unable to access the temporary file", e);
+    }
+    return null;
+  }
+
+  public void deleteFiles() {
+    try {
+      Files.deleteIfExists(USERS_PROPERTIES_FILE_PATH);
+      Files.deleteIfExists(TEMP_FILE_TIMESTAMP_PATH);
+      deletionScheduled = false;
+    } catch (IOException e) {
+      LOGGER.debug("Unable to delete file");
+    }
+  }
+
+  @VisibleForTesting
+  String cronCalculator(Date firstInstall) {
+    Date threeDayTimestamp = new Date(firstInstall.getTime() + TimeUnit.DAYS.toMillis(3));
+    Calendar calendar = Calendar.getInstance();
+    calendar.setTime(threeDayTimestamp);
+
+    int year = calendar.get(Calendar.YEAR);
+    int month = calendar.get(Calendar.MONTH) + 1;
+    int dayOfMonth = calendar.get(Calendar.DAY_OF_MONTH);
+    int hours = calendar.get(Calendar.HOUR_OF_DAY);
+    int minutes = calendar.get(Calendar.MINUTE);
+
+    return String.format("0 %d %d %d %d ? %d", minutes, hours, dayOfMonth, month, year)
+        .replaceAll(" ", "+");
+  }
+
+  public Date installationDate() {
```

Comment:
✏️ This method would become _much_ simpler if you were to take the path of object serialization of the Date.